In [2]:
import pandas as pd
import numpy as np

In [2]:
weather_data = pd.read_csv('west_nile/input/weather.csv')

In [3]:
weather_data.isnull().sum()

Station        0
Date           0
Tmax           0
Tmin           0
Tavg           0
Depart         0
DewPoint       0
WetBulb        0
Heat           0
Cool           0
Sunrise        0
Sunset         0
CodeSum        0
Depth          0
Water1         0
SnowFall       0
PrecipTotal    0
StnPressure    0
SeaLevel       0
ResultSpeed    0
ResultDir      0
AvgSpeed       0
dtype: int64

In [90]:
weather_data.describe()

Station         Tmax         Tmin     DewPoint  ResultSpeed  \
count  2944.000000  2944.000000  2944.000000  2944.000000  2944.000000   
mean      1.500000    76.166101    57.810462    53.457880     6.960666   
std       0.500085    11.461970    10.381939    10.675181     3.587527   
min       1.000000    41.000000    29.000000    22.000000     0.100000   
25%       1.000000    69.000000    50.000000    46.000000     4.300000   
50%       1.500000    78.000000    59.000000    54.000000     6.400000   
75%       2.000000    85.000000    66.000000    62.000000     9.200000   
max       2.000000   104.000000    83.000000    75.000000    24.100000   

         ResultDir  
count  2944.000000  
mean     17.494905  
std      10.063609  
min       1.000000  
25%       7.000000  
50%      19.000000  
75%      25.000000  
max      36.000000

In [91]:
weather_data.replace('M', np.nan, inplace=True)

In [92]:
weather_data.replace('  T', 0.0, inplace=True)

In [93]:
weather_data.isnull().sum()

Station           0
Date              0
Tmax              0
Tmin              0
Tavg             11
Depart         1472
DewPoint          0
WetBulb           4
Heat             11
Cool             11
Sunrise           0
Sunset            0
CodeSum           0
Depth          1472
Water1         2944
SnowFall       1472
PrecipTotal       2
StnPressure       4
SeaLevel          9
ResultSpeed       0
ResultDir         0
AvgSpeed          3
dtype: int64

In [94]:
pd.options.display.max_rows = 999

In [95]:
weather_data

Station        Date  Tmax  Tmin Tavg Depart  DewPoint WetBulb Heat Cool  \
0           1  2007-05-01    83    50   67     14        51      56    0    2   
1           2  2007-05-01    84    52   68    NaN        51      57    0    3   
2           1  2007-05-02    59    42   51     -3        42      47   14    0   
3           2  2007-05-02    60    43   52    NaN        42      47   13    0   
4           1  2007-05-03    66    46   56      2        40      48    9    0   
5           2  2007-05-03    67    48   58    NaN        40      50    7    0   
6           1  2007-05-04    66    49   58      4        41      50    7    0   
7           2  2007-05-04    78    51  NaN    NaN        42      50  NaN  NaN   
8           1  2007-05-05    66    53   60      5        38      49    5    0   
9           2  2007-05-05    66    54   60    NaN        39      50    5    0   
10          1  2007-05-06    68    49   59      4        30      46    6    0   
11          2  2007-05-06    68    52   60    NaN        30      46    5    0   
12          1  2007-05-07    83    47   65     10        41      54    0    0   
13          2  2007-05-07    84    50   67    NaN        39      53    0    2   
14          1  2007-05-08    82    54   68     12        58      62    0    3   
15          2  2007-05-08    80    60   70    NaN        57      63    0    5   
16          1  2007-05-09    77    61   69     13        59      63    0    4   
17          2  2007-05-09    76    63   70    NaN        60      63    0    5   
18          1  2007-05-10    84    56   70     14        52      60    0    5   
19          2  2007-05-10    83    59   71    NaN        52      61    0    6   
20          1  2007-05-11    70    51   61      4        42      51    4    0   
21          2  2007-05-11    73    49   61    NaN        44      51    4    0   
22          1  2007-05-12    64    46   55     -2        36      46   10    0   
23          2  2007-05-12    65    47   56    NaN        37      46    9    0   
24          1  2007-05-13    69    43   56     -2        33      46    9    0   
25          2  2007-05-13    69    44   57    NaN        32      46    8    0   
26          1  2007-05-14    90    56   73     15        47      59    0    8   
27          2  2007-05-14    90    54   72    NaN        45      58    0    7   
28          1  2007-05-15    80    57   69     11        56      61    0    4   
29          2  2007-05-15    82    56   69    NaN        56      61    0    4   
30          1  2007-05-16    66    44   55     -4        43      48   10    0   
31          2  2007-05-16    64    45   55    NaN        42      47   10    0   
32          1  2007-05-17    62    43   53     -6        32      43   12    0   
33          2  2007-05-17    61    44   53    NaN        32      43   12    0   
34          1  2007-05-18    71    42   57     -3        30      46    8    0   
35          2  2007-05-18    70    43   57    NaN        31      47    8    0   
36          1  2007-05-19    80    56   68      8        39      53    0    3   
37          2  2007-05-19    80    56   68    NaN        37      53    0    3   
38          1  2007-05-20    69    43   56     -5        40      44    9    0   
39          2  2007-05-20    69    44   57    NaN        41      45    8    0   
40          1  2007-05-21    80    44   62      1        50      56    3    0   
41          2  2007-05-21    79    46   63    NaN        50      57    2    0   
42          1  2007-05-22    87    55   71     10        46      58    0    6   
43          2  2007-05-22    87    60   74    NaN        44      58    0    9   
44          1  2007-05-23    89    60   75     14        49      61    0   10   
45          2  2007-05-23    89    66   78    NaN        48      61    0   13   
46          1  2007-05-24    88    63   76     14        54      63    0   11   
47          2  2007-05-24    88    66   77    NaN        53      63    0   12   
48          1  2007-05-25    70    56

In [96]:
mask = weather_data['Station'] == 2
station_2 = weather_data[mask]
station_2

Station        Date  Tmax  Tmin Tavg Depart  DewPoint WetBulb Heat Cool  \
1           2  2007-05-01    84    52   68    NaN        51      57    0    3   
3           2  2007-05-02    60    43   52    NaN        42      47   13    0   
5           2  2007-05-03    67    48   58    NaN        40      50    7    0   
7           2  2007-05-04    78    51  NaN    NaN        42      50  NaN  NaN   
9           2  2007-05-05    66    54   60    NaN        39      50    5    0   
11          2  2007-05-06    68    52   60    NaN        30      46    5    0   
13          2  2007-05-07    84    50   67    NaN        39      53    0    2   
15          2  2007-05-08    80    60   70    NaN        57      63    0    5   
17          2  2007-05-09    76    63   70    NaN        60      63    0    5   
19          2  2007-05-10    83    59   71    NaN        52      61    0    6   
21          2  2007-05-11    73    49   61    NaN        44      51    4    0   
23          2  2007-05-12    65    47   56    NaN        37      46    9    0   
25          2  2007-05-13    69    44   57    NaN        32      46    8    0   
27          2  2007-05-14    90    54   72    NaN        45      58    0    7   
29          2  2007-05-15    82    56   69    NaN        56      61    0    4   
31          2  2007-05-16    64    45   55    NaN        42      47   10    0   
33          2  2007-05-17    61    44   53    NaN        32      43   12    0   
35          2  2007-05-18    70    43   57    NaN        31      47    8    0   
37          2  2007-05-19    80    56   68    NaN        37      53    0    3   
39          2  2007-05-20    69    44   57    NaN        41      45    8    0   
41          2  2007-05-21    79    46   63    NaN        50      57    2    0   
43          2  2007-05-22    87    60   74    NaN        44      58    0    9   
45          2  2007-05-23    89    66   78    NaN        48      61    0   13   
47          2  2007-05-24    88    66   77    NaN        53      63    0   12   
49          2  2007-05-25    66    57   62    NaN        46      53    3    0   
51          2  2007-05-26    70    50   60    NaN        55      57    5    0   
53          2  2007-05-27    75    59   67    NaN        49      57    0    2   
55          2  2007-05-28    76    54   65    NaN        49      57    0    0   
57          2  2007-05-29    88    65   77    NaN        59      66    0   12   
59          2  2007-05-30    89    70   80    NaN        57      66    0   15   
61          2  2007-05-31    80    68   74    NaN        61      66    0    9   
63          2  2007-06-01    86    67   77    NaN        63      67    0   12   
65          2  2007-06-02    81    68   75    NaN        62      67    0   10   
67          2  2007-06-03    77    64   71    NaN        60      64    0    6   
69          2  2007-06-04    72    56   64    NaN        57      60    1    0   
71          2  2007-06-05    63    51   57    NaN        47      52    8    0   
73          2  2007-06-06    73    48   61    NaN        47      55    4    0   
75          2  2007-06-07    92    71   82    NaN        61      68    0   17   
77          2  2007-06-08    81    63   72    NaN        54      62    0    7   
79          2  2007-06-09    79    57   68    NaN        49      58    0    3   
81          2  2007-06-10    81    61   71    NaN        50      60    0    6   
83          2  2007-06-11    85    65   75    NaN        51      61    0   10   
85          2  2007-06-12    82    66   74    NaN        51      61    0    9   
87          2  2007-06-13    86    68   77    NaN        53      62    0   12   
89          2  2007-06-14    89    68   79    NaN        54      63    0   14   
91          2  2007-06-15    89    67   78    NaN        56      65    0   13   
93          2  2007-06-16    92    68   80    NaN        61      68    0   15   
95          2  2007-06-17    91    71   81    NaN        65      70    0   16   
97          2  2007-06-18    89    73

In [97]:
station_2['CodeSum'][3]

'BR HZ'

In [98]:
mask = weather_data['Station'] == 1
station_1 = weather_data[mask]
station_1

Station        Date  Tmax  Tmin Tavg Depart  DewPoint WetBulb Heat Cool  \
0           1  2007-05-01    83    50   67     14        51      56    0    2   
2           1  2007-05-02    59    42   51     -3        42      47   14    0   
4           1  2007-05-03    66    46   56      2        40      48    9    0   
6           1  2007-05-04    66    49   58      4        41      50    7    0   
8           1  2007-05-05    66    53   60      5        38      49    5    0   
10          1  2007-05-06    68    49   59      4        30      46    6    0   
12          1  2007-05-07    83    47   65     10        41      54    0    0   
14          1  2007-05-08    82    54   68     12        58      62    0    3   
16          1  2007-05-09    77    61   69     13        59      63    0    4   
18          1  2007-05-10    84    56   70     14        52      60    0    5   
20          1  2007-05-11    70    51   61      4        42      51    4    0   
22          1  2007-05-12    64    46   55     -2        36      46   10    0   
24          1  2007-05-13    69    43   56     -2        33      46    9    0   
26          1  2007-05-14    90    56   73     15        47      59    0    8   
28          1  2007-05-15    80    57   69     11        56      61    0    4   
30          1  2007-05-16    66    44   55     -4        43      48   10    0   
32          1  2007-05-17    62    43   53     -6        32      43   12    0   
34          1  2007-05-18    71    42   57     -3        30      46    8    0   
36          1  2007-05-19    80    56   68      8        39      53    0    3   
38          1  2007-05-20    69    43   56     -5        40      44    9    0   
40          1  2007-05-21    80    44   62      1        50      56    3    0   
42          1  2007-05-22    87    55   71     10        46      58    0    6   
44          1  2007-05-23    89    60   75     14        49      61    0   10   
46          1  2007-05-24    88    63   76     14        54      63    0   11   
48          1  2007-05-25    70    56   63      1        44      53    2    0   
50          1  2007-05-26    65    49   57     -5        55      56    8    0   
52          1  2007-05-27    75    57   66      4        47      56    0    1   
54          1  2007-05-28    79    50   65      2        47      56    0    0   
56          1  2007-05-29    88    60   74     10        58      65    0    9   
58          1  2007-05-30    87    66   77     13        58      65    0   12   
60          1  2007-05-31    78    65   72      8        61      65    0    7   
62          1  2007-06-01    85    62   74     10        63      66    0    9   
64          1  2007-06-02    81    65   73      9        63      67    0    8   
66          1  2007-06-03    78    62   70      6        60      63    0    5   
68          1  2007-06-04    71    58   65      1        58      60    0    0   
70          1  2007-06-05    64    47   56     -9        48      51    9    0   
72          1  2007-06-06    71    44   58     -8        48      55    7    0   
74          1  2007-06-07    91    71   81     15        62      69    0   16   
76          1  2007-06-08    81    57   69      3        53      61    0    4   
78          1  2007-06-09    79    52   66      0        47      57    0    1   
80          1  2007-06-10    81    55   68      1        51      59    0    3   
82          1  2007-06-11    86    58   72      5        52      61    0    7   
84          1  2007-06-12    83    62   73      5        50      60    0    8   
86          1  2007-06-13    87    60   74      6        53      62    0    9   
88          1  2007-06-14    89    64   77      9        55      64    0   12   
90          1  2007-06-15    91    63   77      9        56      64    0   12   
92          1  2007-06-16    93    66   80     11        60      67    0   15   
94          1  2007-06-17    91    70   81     12        64      70    0   16   
96          1  2007-06-18    89    69

In [ ]:
station_1.info()

Station        1472 non-null int64

Date           1472 non-null object #May need to convert this to whatever we are going to use for the other data

Tmax           1472 non-null int64

Tmin           1472 non-null int64

Tavg           1472 non-null object # Convert to int

Depart         1472 non-null object # Convert to int

DewPoint       1472 non-null int64

WetBulb        1469 non-null object # Convert to int

Heat           1472 non-null object # Convert to int

Cool           1472 non-null object # Convert to int

Sunrise        1472 non-null object # Convert to int

Sunset         1472 non-null object # Convert to int Use these 2 to make new feature

CodeSum        1472 non-null object # need to grab code(s). Countvectorize?

Depth          1472 non-null object # Convert to int, might all be zeros

Water1         0 non-null float64   #Remove column, all null

SnowFall       1472 non-null object # Convert to int what to do with T?

PrecipTotal    1472 non-null object # Convert to int what to do with T?

StnPressure    1470 non-null object # Convert to int

SeaLevel       1467 non-null object # Convert to int

ResultSpeed    1472 non-null float64

ResultDir      1472 non-null int64

AvgSpeed       1472 non-null object # Convert to int

dtypes: float64(2), int64(5), object(15)

memory usage: 304.5+ KB

In [ ]:
weather_data['PrecipTotal'].astype('float64')

In [ ]:
mask = station_1['PrecipTotal'] == 'T'
station_1_depth = station_1[mask]
station_1_depth

In [100]:
station_2.isnull().sum()

Station           0
Date              0
Tmax              0
Tmin              0
Tavg             11
Depart         1472
DewPoint          0
WetBulb           1
Heat             11
Cool             11
Sunrise           0
Sunset            0
CodeSum           0
Depth          1472
Water1         1472
SnowFall       1472
PrecipTotal       2
StnPressure       2
SeaLevel          4
ResultSpeed       0
ResultDir         0
AvgSpeed          3
dtype: int64

In [4]:
spray_data = pd.read_csv('west_nile/input/spray.csv')

In [29]:
spray_count = []
for i in range(len(train_data['Date'])):
    count = 0
    for j in range(len(spray_data['Date'])):
        if spray_data['Date'][j] == train_data['Date'][i]:
            if distance_finder(spray_data["Latitude"][j], spray_data["Longitude"][j], train_data['Latitude'][i], train_data['Longitude'][i]) < 100:
                count +=1
            else:
                pass
    spray_count.append(count)            

In [30]:
len(spray_count)

10506

In [33]:
train_data['spray_count'] = spray_count

In [39]:
mask = train_data['spray_count'] > 0
spray_more = train_data[mask]
spray_more

Date                                            Address  \
9177  2013-07-25  3900 North Springfield Avenue, Chicago, IL 606...   
9408  2013-08-08  2800 North Francisco Avenue, Chicago, IL 60618...   
9590  2013-08-15  7000 West Armitage Avenue, Chicago, IL 60707, USA   
9591  2013-08-15  7000 West Armitage Avenue, Chicago, IL 60707, USA   
9691  2013-08-15  1300 North Laramie Avenue, Chicago, IL 60651, USA   
9892  2013-08-29  5800 North Western Avenue, Chicago, IL 60659, USA   
9893  2013-08-29  5800 North Western Avenue, Chicago, IL 60659, USA   
9894  2013-08-29  5800 North Western Avenue, Chicago, IL 60659, USA   

                     Species  Block               Street  Trap  \
9177  CULEX PIPIENS/RESTUANS     39    N SPRINGFIELD AVE  T228   
9408           CULEX PIPIENS     28      N FRANCISCO AVE  T147   
9590  CULEX PIPIENS/RESTUANS     70    W ARMITAGE AVENUE  T151   
9591           CULEX PIPIENS     70    W ARMITAGE AVENUE  T151   
9691  CULEX PIPIENS/RESTUANS     13        N LARAMIE AVE  T230   
9892  CULEX PIPIENS/RESTUANS     58        N WESTERN AVE  T028   
9893          CULEX RESTUANS     58        N WESTERN AVE  T028   
9894           CULEX PIPIENS     58        N WESTERN AVE  T028   

                     AddressNumberAndStreet   Latitude  Longitude  \
9177   3900  N SPRINGFIELD AVE, Chicago, IL  41.951866 -87.725057   
9408     2800  N FRANCISCO AVE, Chicago, IL  41.932094 -87.700117   
9590  7000   W ARMITAGE AVENUE, Chicago, IL  41.916265 -87.800515   
9591  7000   W ARMITAGE AVENUE, Chicago, IL  41.916265 -87.800515   
9691       1300  N LARAMIE AVE, Chicago, IL  41.904194 -87.756155   
9892       5800  N WESTERN AVE, Chicago, IL  41.986921 -87.689778   
9893       5800  N WESTERN AVE, Chicago, IL  41.986921 -87.689778   
9894       5800  N WESTERN AVE, Chicago, IL  41.986921 -87.689778   

      AddressAccuracy  NumMosquitos  WnvPresent  spray_count  
9177                8            24           0           14  
9408                8             5           0            3  
9590                8            11           0            2  
9591                8             9           1            2  
9691                9            15           1            3  
9892                9             3           0            2  
9893                9             2           0            2  
9894                9            17           1            2

In [136]:
spray_data.isnull().sum()

Date           0
Time         584
Latitude       0
Longitude      0
dtype: int64

In [105]:
spray_data.describe()

Latitude     Longitude
count  14835.000000  14835.000000
mean      41.904828    -87.736690
std        0.104381      0.067292
min       41.713925    -88.096468
25%       41.785001    -87.794225
50%       41.940075    -87.727853
75%       41.980978    -87.694108
max       42.395983    -87.586727

In [106]:
spray_data.head()

Date        Time   Latitude  Longitude
0  2011-08-29  6:56:58 PM  42.391623 -88.089163
1  2011-08-29  6:57:08 PM  42.391348 -88.089163
2  2011-08-29  6:57:18 PM  42.391022 -88.089157
3  2011-08-29  6:57:28 PM  42.390637 -88.089158
4  2011-08-29  6:57:38 PM  42.390410 -88.088858

In [6]:
train_data = pd.read_csv('train.csv')

In [30]:
test_data = pd.read_csv('test.csv')

In [109]:
train_data.describe()

Block      Latitude     Longitude  AddressAccuracy  \
count  10506.000000  10506.000000  10506.000000     10506.000000   
mean      35.687797     41.841139    -87.699908         7.819532   
std       24.339468      0.112742      0.096514         1.452921   
min       10.000000     41.644612    -87.930995         3.000000   
25%       12.000000     41.732984    -87.760070         8.000000   
50%       33.000000     41.846283    -87.694991         8.000000   
75%       52.000000     41.954690    -87.627796         9.000000   
max       98.000000     42.017430    -87.531635         9.000000   

       NumMosquitos    WnvPresent  
count  10506.000000  10506.000000  
mean      12.853512      0.052446  
std       16.133816      0.222936  
min        1.000000      0.000000  
25%        2.000000      0.000000  
50%        5.000000      0.000000  
75%       17.000000      0.000000  
max       50.000000      1.000000

In [110]:
train_data.shape

(10506, 12)

In [135]:
np.unique(spray_data['Date'])

array(['2011-08-29', '2011-09-07', '2013-07-17', '2013-07-25',
       '2013-08-08', '2013-08-15', '2013-08-16', '2013-08-22',
       '2013-08-29', '2013-09-05'], dtype=object)

In [50]:
train_data['timestamp'] = pd.to_datetime(train_data['Date'])

In [51]:
spray_data['timestamp'] = pd.to_datetime(spray_data['Date'])

## Creating Dummy for traps with spray data 

This is my attempt to create dummy variables that would show which traps the spraying passed. Results showed that the data was a little sparce to do analysis on.

In [17]:
from math import sin, cos, sqrt, atan2, radians


def distance_finder(lat1, lon1, lat2, lon2):
# approximate radius of earth in m
    R = 6378100.0

    lat1 = radians(lat1)
    lon1 = radians(lon1)
    lat2 = radians(lat2)
    lon2 = radians(lon2)
    
    dlon = lon2 - lon1
    dlat = lat2 - lat1
    a = sin(dlat / 2)**2 + cos(lat1) * cos(lat2) * sin(dlon / 2)**2
    c = 2 * atan2(sqrt(a), sqrt(1 - a))

    distance = R * c

    return distance

In [72]:
spray_count_day_0 = []
for i in range(len(train_data['Date'])):
    count = 0
    for j in range(len(spray_data['Date'])):
        if spray_data['Date'][j] == train_data['Date'][i]:
            if distance_finder(spray_data["Latitude"][j], spray_data["Longitude"][j], train_data['Latitude'][i], train_data['Longitude'][i]) < 100:
                count +=1
            else:
                pass
    spray_count_day_0.append(count)            

In [75]:
train_data['spray_count_day_0'] = spray_count_day_0

In [73]:
import datetime
spray_count_day_1 = []
for i in range(len(train_data['Date'])):
    count = 0
    for j in range(len(spray_data['Date'])):
        if (spray_data['timestamp'][j] + datetime.timedelta(days=1))  == train_data['timestamp'][i]:
            if distance_finder(spray_data["Latitude"][j], spray_data["Longitude"][j], train_data['Latitude'][i], train_data['Longitude'][i]) < 100:
                count +=1
            else:
                pass
    spray_count_day_1.append(count)            

In [79]:
train_data['spray_count_day_1'] = spray_count_day_1

In [74]:
import datetime
spray_count_day_2 = []
for i in range(len(train_data['Date'])):
    count = 0
    for j in range(len(spray_data['Date'])):
        if (spray_data['timestamp'][j] + datetime.timedelta(days=2))  == train_data['timestamp'][i]:
            if distance_finder(spray_data["Latitude"][j], spray_data["Longitude"][j], train_data['Latitude'][i], train_data['Longitude'][i]) < 100:
                count +=1
            else:
                pass
    spray_count_day_2.append(count)            

In [77]:
train_data['spray_count_day_2'] = spray_count_day_2

In [80]:
train_data

Date                                            Address  \
0      2007-05-29  4100 North Oak Park Avenue, Chicago, IL 60634,...   
1      2007-05-29  4100 North Oak Park Avenue, Chicago, IL 60634,...   
2      2007-05-29  6200 North Mandell Avenue, Chicago, IL 60646, USA   
3      2007-05-29    7900 West Foster Avenue, Chicago, IL 60656, USA   
4      2007-05-29    7900 West Foster Avenue, Chicago, IL 60656, USA   
5      2007-05-29   1500 West Webster Avenue, Chicago, IL 60614, USA   
6      2007-05-29     2500 West Grand Avenue, Chicago, IL 60654, USA   
7      2007-05-29        1100 Roosevelt Road, Chicago, IL 60608, USA   
8      2007-05-29        1100 Roosevelt Road, Chicago, IL 60608, USA   
9      2007-05-29   1100 West Chicago Avenue, Chicago, IL 60642, USA   
10     2007-05-29    2100 North Stave Street, Chicago, IL 60647, USA   
11     2007-05-29    2200 North Cannon Drive, Chicago, IL 60614, USA   
12     2007-05-29    2200 North Cannon Drive, Chicago, IL 60614, USA   
13     2007-05-29     2200 West 113th Street, Chicago, IL 60643, USA   
14     2007-05-29     2200 West 113th Street, Chicago, IL 60643, USA   
15     2007-05-29   1100 South Peoria Street, Chicago, IL 60608, USA   
16     2007-05-29      1700 West 95th Street, Chicago, IL 60643, USA   
17     2007-05-29      2200 West 89th Street, Chicago, IL 60643, USA   
18     2007-05-29      2200 West 89th Street, Chicago, IL 60643, USA   
19     2007-05-29       North Streeter Drive, Chicago, IL 60611, USA   
20     2007-05-29       North Streeter Drive, Chicago, IL 60611, USA   
21     2007-05-29  6500 North Oak Park Avenue, Chicago, IL 60631,...   
22     2007-05-29   7500 North Oakley Avenue, Chicago, IL 60645, USA   
23     2007-05-29     1500 North Long Avenue, Chicago, IL 60651, USA   
24     2007-05-29  8900 South Carpenter Street, Chicago, IL 60620...   
25     2007-06-05  4100 North Oak Park Avenue, Chicago, IL 60634,...   
26     2007-06-05  4100 North Oak Park Avenue, Chicago, IL 60634,...   
27     2007-06-05  4100 North Oak Park Avenue, Chicago, IL 60634,...   
28     2007-06-05    7900 West Foster Avenue, Chicago, IL 60656, USA   
29     2007-06-05    7900 West Foster Avenue, Chicago, IL 60656, USA   
...           ...                                                ...   
10476  2013-09-26       South Cottage Grove Avenue, Chicago, IL, USA   
10477  2013-09-26       South Cottage Grove Avenue, Chicago, IL, USA   
10478  2013-09-26       South Cottage Grove Avenue, Chicago, IL, USA   
10479  2013-09-26    5800 North Pulaski Road, Chicago, IL 60646, USA   
10480  2013-09-26     4000 East 130th Street, Chicago, IL 60633, USA   
10481  2013-09-26     4000 East 130th Street, Chicago, IL 60633, USA   
10482  2013-09-26     4000 East 130th Street, Chicago, IL 60633, USA   
10483  2013-09-26    9100 West Higgins Road, Rosemont, IL 60018, USA   
10484  2013-09-26  ORD Terminal 5, O'Hare International Airport, ...   
10485  2013-09-26  ORD Terminal 5, O'Hare International Airport, ...   
10486  2013-09-26  ORD Terminal 5, O'Hare International Airport, ...   
10487  2013-09-26  ORD Terminal 5, O'Hare International Airport, ...   
10488  2013-09-26  ORD Terminal 5, O'Hare International Airport, ...   
10489  2013-09-26  ORD Terminal 5, O'Hare International Airport, ...   
10490  2013-09-26   4800 West Montana Street, Chicago, IL 60639, USA   
10491  2013-09-26  5100 North Mont Clare Avenue, Chicago, IL 6065...   
10492  2013-09-26  5100 North Mont Clare Avenue, Chicago, IL 6065...   
10493  2013-09-26  8200 South Kostner Avenue, Chicago, IL 60652, USA   
10494  2013-09-26                  East 91st Place, Chicago, IL, USA   
10495  2013-09-26                  East 91st Place, Chicago, IL, USA   
10496  2013-09-26   1700 West Addison Street, Chicago, IL 60613, USA   
10497  2013-09-26          West Garfield Boulevard, Chicago, IL, USA   
10498  2013-09-26  1300 North Laramie Avenue, Chicago, IL 60651, USA   
10499  2013-09-26  1300 North Laramie Avenue, Chicago

In [82]:
new_train.to_csv("./train_w_spray.csv")

In [86]:
spray_data

Date        Time   Latitude  Longitude
0      2011-08-29  6:56:58 PM  42.391623 -88.089163
1      2011-08-29  6:57:08 PM  42.391348 -88.089163
2      2011-08-29  6:57:18 PM  42.391022 -88.089157
3      2011-08-29  6:57:28 PM  42.390637 -88.089158
4      2011-08-29  6:57:38 PM  42.390410 -88.088858
5      2011-08-29  6:57:48 PM  42.390395 -88.088315
6      2011-08-29  6:57:58 PM  42.390673 -88.088002
7      2011-08-29  6:58:08 PM  42.391027 -88.088002
8      2011-08-29  6:58:18 PM  42.391403 -88.088003
9      2011-08-29  6:58:28 PM  42.391718 -88.087995
10     2011-08-29  6:58:38 PM  42.392038 -88.087988
11     2011-08-29  6:58:48 PM  42.392410 -88.088078
12     2011-08-29  6:58:58 PM  42.392800 -88.088332
13     2011-08-29  6:59:08 PM  42.393257 -88.088370
14     2011-08-29  6:59:18 PM  42.393600 -88.088232
15     2011-08-29  7:00:08 PM  42.393203 -88.088417
16     2011-08-29  7:00:18 PM  42.392792 -88.088312
17     2011-08-29  7:00:28 PM  42.392477 -88.088210
18     2011-08-29  7:00:38 PM  42.392385 -88.088697
19     2011-08-29  7:00:48 PM  42.392098 -88.089113
20     2011-08-29  7:00:58 PM  42.392180 -88.089498
21     2011-08-29  7:01:08 PM  42.392428 -88.089790
22     2011-08-29  7:01:18 PM  42.392722 -88.090042
23     2011-08-29  7:01:28 PM  42.392958 -88.090300
24     2011-08-29  7:01:38 PM  42.392990 -88.090895
25     2011-08-29  7:01:48 PM  42.393002 -88.091612
26     2011-08-29  7:01:58 PM  42.393008 -88.092010
27     2011-08-29  7:02:08 PM  42.393020 -88.092548
28     2011-08-29  7:02:18 PM  42.393103 -88.093182
29     2011-08-29  7:02:28 PM  42.393118 -88.093737
30     2011-08-29  7:02:38 PM  42.393423 -88.093895
31     2011-08-29  7:02:48 PM  42.393743 -88.094075
32     2011-08-29  7:02:58 PM  42.394105 -88.094250
33     2011-08-29  7:03:08 PM  42.394478 -88.094223
34     2011-08-29  7:03:18 PM  42.394860 -88.094198
35     2011-08-29  7:03:28 PM  42.395085 -88.094515
36     2011-08-29  7:03:38 PM  42.395095 -88.095027
37     2011-08-29  7:03:48 PM  42.395078 -88.095520
38     2011-08-29  7:03:58 PM  42.395352 -88.095772
39     2011-08-29  7:04:08 PM  42.395697 -88.095833
40     2011-08-29  7:04:18 PM  42.395983 -88.095757
41     2011-08-29  7:04:48 PM  42.395218 -88.095685
42     2011-08-29  7:04:58 PM  42.394932 -88.095555
43     2011-08-29  7:05:08 PM  42.394617 -88.095597
44     2011-08-29  7:05:18 PM  42.394290 -88.095600
45     2011-08-29  7:05:28 PM  42.393957 -88.095485
46     2011-08-29  7:05:38 PM  42.393673 -88.095372
47     2011-08-29  7:05:48 PM  42.393332 -88.095222
48     2011-08-29  7:05:58 PM  42.393037 -88.095143
49     2011-08-29  7:06:08 PM  42.392647 -88.095138
50     2011-08-29  7:06:18 PM  42.392348 -88.095140
51     2011-08-29  7:06:28 PM  42.392237 -88.095455
52     2011-08-29  7:06:38 PM  42.392247 -88.095980
53     2011-08-29  7:06:48 PM  42.392463 -88.096435
54     2011-08-29  7:06:58 PM  42.392838 -88.096445
55     2011-08-29  7:07:08 PM  42.393193 -88.096447
56     2011-08-29  7:07:18 PM  42.393488 -88.096267
57     2011-08-29  7:07:28 PM  42.393603 -88.095917
58     2011-08-29  7:07:38 PM  42.393678 -88.095593
59     2011-08-29  7:08:18 PM  42.392527 -88.095148
60     2011-08-29  7:08:28 PM  42.392197 -88.095163
61     2011-08-29  7:08:38 PM  42.391860 -88.095140
62     2011-08-29  7:08:48 PM  42.391518 -88.095143
63     2011-08-29  7:08:58 PM  42.391297 -88.095203
64     2011-08-29  7:09:08 PM  42.391275 -88.095608
65     2011-08-29  7:09:18 PM  42.391270 -88.096035
66     2011-08-29  7:09:28 PM  42.391238 -88.096468
67     2011-08-29  7:09:38 PM  42.390952 -88.096437
68     2011-08-29  7:09:48 PM  42.390642 -88.096440
69     2011-08-29  7:09:58 PM  42.390335 -88.096443
70     2011-08-29  7:10:08 PM  42.390048 -88.096435
71     2011-08-29  7:10:18 PM  42.389870 -88.096258
72     2011-08-29  7:10:28 PM  42.389867 -88.095878
73     2011-08-29  7:10:38 PM  42.389858 -88.095487
74     2011-08-29  7:10:48 PM  42.389855 -88.095080
75     2011-08-29  7:10:58 PM  42.389843 -88.094667
76     20

In [7]:
new_train = pd.read_csv('new_train.csv')

In [6]:
pd.options.display.max_columns = 100

# Combineing Training rows

This is the fucntion I created in order to combine rows in the taining data based on the date, trap, and species so that the the limit of 50 in the trap would not skew the results

In [64]:
def combine_date_trap(pan):
    d = pan.groupby(['Date', 'Trap', 'Species']).sum()
    d = d.reset_index()
    s = pan.drop_duplicates(['Date', 'Trap', 'Species'])
    s = s.drop(["NumMosquitos", 'WnvPresent'], axis =1)
    s = s.reset_index()
    s['NumMosquitos'] = d['NumMosquitos']
    s['WnvPresent'] = d['WnvPresent']
    return s

In [65]:
new_train_combine = combine_date_trap(new_train)

In [67]:
new_train_combine.to_csv('./new_train_combine_3.csv')

In [75]:
new_train_combine

Unnamed: 0        Date  \
0               0  2007-05-29   
2               2  2007-05-29   
3               3  2007-05-29   
5               5  2007-05-29   
6               6  2007-05-29   
7               7  2007-05-29   
9               9  2007-05-29   
10             10  2007-05-29   
11             11  2007-05-29   
13             13  2007-05-29   
15             15  2007-05-29   
16             16  2007-05-29   
17             17  2007-05-29   
19             19  2007-05-29   
21             21  2007-05-29   
22             22  2007-05-29   
23             23  2007-05-29   
24             24  2007-05-29   
25             25  2007-06-05   
28             28  2007-06-05   
30             30  2007-06-05   
33             33  2007-06-05   
34             34  2007-06-05   
37             37  2007-06-05   
38             38  2007-06-05   
40             40  2007-06-05   
41             41  2007-06-05   
42             42  2007-06-05   
43             43  2007-06-05   
46             46  2007-06-05   
...           ...         ...   
10456       10456  2013-09-26   
10457       10457  2013-09-26   
10459       10459  2013-09-26   
10460       10460  2013-09-26   
10462       10462  2013-09-26   
10463       10463  2013-09-26   
10464       10464  2013-09-26   
10466       10466  2013-09-26   
10468       10468  2013-09-26   
10471       10471  2013-09-26   
10473       10473  2013-09-26   
10475       10475  2013-09-26   
10476       10476  2013-09-26   
10479       10479  2013-09-26   
10480       10480  2013-09-26   
10483       10483  2013-09-26   
10484       10484  2013-09-26   
10490       10490  2013-09-26   
10491       10491  2013-09-26   
10493       10493  2013-09-26   
10494       10494  2013-09-26   
10496       10496  2013-09-26   
10497       10497  2013-09-26   
10498       10498  2013-09-26   
10500       10500  2013-09-26   
10501       10501  2013-09-26   
10502       10502  2013-09-26   
10503       10503  2013-09-26   
10504       10504  2013-09-26   
10505       10505  2013-09-26   

                                                 Address  \
0      4100 North Oak Park Avenue, Chicago, IL 60634,...   
2      6200 North Mandell Avenue, Chicago, IL 60646, USA   
3        7900 West Foster Avenue, Chicago, IL 60656, USA   
5       1500 West Webster Avenue, Chicago, IL 60614, USA   
6         2500 West Grand Avenue, Chicago, IL 60654, USA   
7            1100 Roosevelt Road, Chicago, IL 60608, USA   
9       1100 West Chicago Avenue, Chicago, IL 60642, USA   
10       2100 North Stave Street, Chicago, IL 60647, USA   
11       2200 North Cannon Drive, Chicago, IL 60614, USA   
13        2200 West 113th Street, Chicago, IL 60643, USA   
15      1100 South Peoria Street, Chicago, IL 60608, USA   
16         1700 West 95th Street, Chicago, IL 60643, USA   
17         2200 West 89th Street, Chicago, IL 60643, USA   
19          North Streeter Drive, Chicago, IL 60611, USA   
21     6500 North Oak Park Avenue, Chicago, IL 60631,...   
22      7500 North Oakley Avenue, Chicago, IL 60645, USA   
23        1500 North Long Avenue, Chicago, IL 60651, USA   
24     8900 South Carpenter Street, Chicago, IL 60620...   
25     4100 North Oak Park Avenue, Chicago, IL 60634,...   
28       7900 West Foster Avenue, Chicago, IL 60656, USA   
30      1500 West Webster Avenue, Chicago, IL 60614, USA   
33        2500 West Grand Avenue, Chicago, IL 60654, USA   
34           1100 Roosevelt Road, Chicago, IL 60608, USA   
37      1100 West Chicago Avenue, Chicago, IL 60642, USA   
38       2100 North Stave Street, Chicago, IL 60647, USA   
40       2200 North Cannon Drive, Chicago, IL 60614, USA   
41        2200 West 113th Street, Chicago, IL 60643, USA   
42         1700 West 95th Street, Chicago, IL 60643, USA   
43         2200 West 89th Street, Chicago, IL 60643, USA   
46     6500 North Oak Park Avenue, Chicago, IL 60631,...   
...                                                  ...   
10456  7100 South South Shore Drive, Chi

In [77]:
new_train_combine.to_csv('./new_train_combine.csv')